pep-0657.rst

```diff
@@ -210,11 +210,17 @@ Offset semantics
 ^^^^^^^^^^^^^^^^
 
 These offsets are propagated by the compiler from the ones stored currently in
-all AST nodes. They are 1-indexed and a value of 0 will mean that the
-information is not available. Although the AST nodes use ``int`` types to store
-these values, ``uint8_t`` types will be used for storage in the new map to
-minimize storage impact. This decision allows offsets to go from 0 to 255,
-while offsets bigger than these values will be treated as missing (value of 0).
+all AST nodes. The output of the public APIs (``co_positions`` and ``PyCode_Addr2Location``)
+that deal with these attributes use 0-indexed offsets (just like the AST nodes), but the underlying
+implementation is free to represent the actual data in whatever form they choose to be most efficient.
+The error code regarding information not available is ``None`` for the ``co_positions()`` API,
+and ``-1`` for the ``PyCode_Addr2Location`` API. The availability of the information highly depends
+on whether the offsets fall under the range, as well as the runtime flags for the interpreter
+configuration.
+
+Although the AST nodes use ``int`` types to store these values, ``uint8_t`` types will be used
+for storage in the new map to minimize storage impact. This decision allows offsets to go from
+0 to 255, while offsets bigger than these values will be treated as missing (value of 0).
```

Comment:
If 0 is used as an actual offset for the first char of the line, it cannot also be used as missing.  255 could be used to mean 255 or more, to be turned into None.  Or am I missing something?

Reply:
In the public API missing will return -1 in C and None in Python. The internal values are an implementation detail